src/main/java/frc/team997/lib/Trajectory/AsymmetricTrapezoidProfile.java

```diff
@@ -0,0 +1,355 @@
+    public final Constraints constraints;
+    public final State initial;
+    public final State target;
+    public final double direction;
+
+    public static class Constraints {
+        final double maxVelocity;
+        final double maxAcceleration;
+        final double maxDeceleration;
+        /**
+         * Construct constraints for a AsymmetricTrapezoidProfile.
+         *
+         * @param maxVelocity maximum velocity.
+         * @param maxAcceleration maximum acceleration.
+         * @param maxDeceleration maximum deceleration.
+         */
+        public Constraints(double maxVelocity, double maxAcceleration, double maxDeceleration) {
+            this.maxVelocity = maxVelocity;
+            this.maxAcceleration = maxAcceleration;
+            this.maxDeceleration = maxDeceleration;
+        }
+        @Override
+        public boolean equals(Object other) {
+            double epsilon = 0.0001;
+            if (other instanceof Constraints) {
+                Constraints rhs = (Constraints) other;
+                return Math.abs(this.maxVelocity - rhs.maxVelocity) < epsilon
+                        && Math.abs(this.maxAcceleration - rhs.maxAcceleration) < epsilon
+                        && Math.abs(this.maxDeceleration - rhs.maxDeceleration) < epsilon;
+            } else {
+                return false;
+            }
+        }
+        @Override
+        public int hashCode() {
+            return Objects.hash(maxVelocity, maxAcceleration, maxDeceleration);
+        }
+        @Override
+        public String toString() {
+            return "Constraints[maxVelocity: "+maxVelocity+", maxAcceleration: "+maxAcceleration+", maxDeceleration"+maxDeceleration+"]";
+        }
+    }
+    public static class State {
+        final double position;
+        final double velocity;
+        /**
+         * Construct a state for a AsymmetricTrapezoidProfile.
+         *
+         * @param position state position.
+         * @param velocity state velocity.
+         */
+        public State(double position, double velocity) {
+            this.position = position;
+            this.velocity = velocity;
+        }
+        @Override
+        public boolean equals(Object other) {
+            double epsilon = 0.0001;
+            if (other instanceof State) {
+                State rhs = (State) other;
+                return Math.abs(this.position - rhs.position) < epsilon
+                        && Math.abs(this.velocity - rhs.velocity) < epsilon;
+            } else {
+                return false;
+            }
+        }
+        @Override
+        public int hashCode() {
+            return Objects.hash(position, velocity);
+        }
+        @Override
+        public String toString() {
+            return "State[position: "+position+", velocity: "+velocity+"]";
+        }
+    }
+
+    /**
+     * A data class for profile phase data.
+     */
+    public static class Phase {
```

Comment:
This is a good abstraction-- could be useful for non-calculated profiles (accelerate this far for this long).
Maybe move it to its own class so that can be cleaner? (with a more descriptive name)